sphinx_immaterial/python_domain_fixes.py

```diff
@@ -197,6 +206,481 @@ def wrapper(self: GoogleDocstring, section: str) -> List[str]:
+            break
+
+    for name_node, param_node in replacements:
+        name = name_node.astext()
+        refnode = sphinx.addnodes.pending_xref(
+            "",
+            name_node.deepcopy(),
+            refdomain="py",
+            reftype="parameter",
+            reftarget=f"{symbol}.{name}",
+            refwarn=False,
+        )
+        name_node.replace_self(refnode)
+
+    return sig_param_nodes
+
+
+def _add_parameter_documentation_ids(
+    directive: sphinx.domains.python.PyObject,
+    env: sphinx.environment.BuildEnvironment,
+    obj_content: sphinx.addnodes.desc_content,
+    sig_param_nodes_for_signature: List[Dict[str, docutils.nodes.Element]],
+    symbols: List[str],
+) -> None:
+
+    qualify_parameter_ids = env.config.python_qualify_parameter_ids
+
+    param_options = apidoc_formatting.get_object_description_options(
+        env, "py", "parameter"
+    )
+
+    py = cast(sphinx.domains.python.PythonDomain, env.get_domain("py"))
+
+    def cross_link_single_parameter(
+        param_name: str, param_node: docutils.nodes.term
+    ) -> None:
+        # Determine the number of unique declarations of this parameter.
+        #
+        # If this single object description has multiple signatures, the same
+        # parameter name may be declared in more than one of those signatures.
+        # In the parameter description, we will replace the bare parameter name
+        # with the list of all of the distinct declarations of the parameter.
+        # Identical declarations in more than one signature will only be
+        # included once.
+        unique_decls: Dict[str, Tuple[int, docutils.nodes.Element]] = {}
+        for i, sig_param_nodes in enumerate(sig_param_nodes_for_signature):
+            desc_param_node = sig_param_nodes.get(param_name)
+            if desc_param_node is None:
+                continue
+            decl_text = desc_param_node.astext().strip()
+            unique_decls.setdefault(decl_text, (i, desc_param_node))
+        if not unique_decls:
+            all_params = {}
+            for sig_param_nodes in sig_param_nodes_for_signature:
+                all_params.update(sig_param_nodes)
+            logger.warning(
+                "Parameter name %r does not match any of the parameters "
+                "defined in the signature: %r",
+                param_name,
+                list(all_params.keys()),
+                location=param_node,
+            )
+            return
+
+        generate_synopses = param_options["generate_synopses"]
+        if generate_synopses is not None:
+            synopsis = sphinx_utils.summarize_element_text(
+                param_node.parent[-1], generate_synopses
+            )
+
+        unqualified_param_id = f"p-{param_name}"
+
+        # Set ids of the parameter node.
+        for symbol_i, _ in unique_decls.values():
+            symbol = symbols[symbol_i]
+            param_symbol = f"{symbol}.{param_name}"
+
+            if synopsis:
```

Comment:
I get an error when trying to disable synopses
```
Exception occurred:
  File "C:\Users\ytreh\Documents\GitHub\venv\lib\site-packages\sphinx_immaterial\python_domain_fixes.py", line 489, in cross_link_single_parameter
    if synopsis:
UnboundLocalError: local variable 'synopsis' referenced before assignment
```
using the config:
```py
object_description_options = [
    ("py:parameter", dict(generate_synopses=None)),
]
```
So, maybe do
```suggestion
            if generate_synopses is not None and synopsis:
```

Reply:
Thanks --- fixed.